internals/daemon/api_checks.go

```diff
@@ -66,3 +70,49 @@ func v1GetChecks(c *Command, r *http.Request, _ *UserState) Response {
 	}
 	return SyncResponse(infos)
 }
+
+func v1PostChecks(c *Command, r *http.Request, _ *UserState) Response {
+	var payload struct {
+		Action string   `json:"action"`
+		Checks []string `json:"checks"`
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode data from request body: %v", err)
+	}
+
+	if len(payload.Checks) == 0 {
+		return BadRequest("must specify checks for %s action", payload.Action)
+	}
+
+	var err error
+	var checks []string
```

Comment:
Nit: I'd probably move these to just above the `switch` block, as they're not needed before that.